SU2_CFD/src/solvers/CDiscAdjSolver.cpp

```diff
@@ -297,32 +297,29 @@ void CDiscAdjSolver::RegisterOutput(CGeometry *geometry, CConfig *config) {
   direct_solver->GetNodes()->RegisterSolution(false);
 }
 
-void CDiscAdjSolver::ExtractAdjoint_Solution(CGeometry *geometry, CConfig *config, bool CrossTerm){
+void CDiscAdjSolver::ExtractAdjoint_Solution(CGeometry *geometry, CConfig *config, bool CrossTerm) {
 
   const bool time_n1_needed = config->GetTime_Marching() == TIME_MARCHING::DT_STEPPING_2ND;
   const bool time_n_needed = (config->GetTime_Marching() == TIME_MARCHING::DT_STEPPING_1ST) || time_n1_needed;
 
   const su2double relax = (config->GetInnerIter()==0) ? 1.0 : config->GetRelaxation_Factor_Adjoint();
 
-  su2double Solution[MAXNVAR] = {0.0};
-
-  /*--- Set Residuals to zero ---*/
-
-  SetResToZero();
+  /*--- Thread-local residual variables. ---*/
 
   su2double resMax[MAXNVAR] = {0.0}, resRMS[MAXNVAR] = {0.0};
   const su2double* coordMax[MAXNVAR] = {nullptr};
   unsigned long idxMax[MAXNVAR] = {0};
 
   /*--- Set the old solution and compute residuals. ---*/
 
-  if(!config->GetMultizone_Problem()) nodes->Set_OldSolution();
+  if (!config->GetMultizone_Problem()) nodes->Set_OldSolution();
 
   SU2_OMP_FOR_STAT(omp_chunk_size)
   for (auto iPoint = 0ul; iPoint < nPoint; iPoint++) {
 
     /*--- Extract the adjoint solution ---*/
 
+    su2double Solution[MAXNVAR] = {0.0};
```

Comment:
moving that container was necessary to have one per thread, right? so it wasn't correct for potential omp adjoint before?

Reply:
no, just making it more local, same idea as iPoint.